python/lsst/ts/wep/utils/zernikeUtils.py

```diff
@@ -515,6 +512,75 @@ def convertZernikesToPsfWidth(
     return dFWHM
 
 
+def correctAOSResid(aos_resid: float | np.ndarray) -> float | np.ndarray:
+    """Correct the AOS residual for over-estimation.
+
+    This correction is empirically fit in
+    https://gist.github.com/jfcrenshaw/24056516cfa3ce0237e39507674a43e1
+
+    Parameters
+    ----------
+    aos_resid : float or np.ndarray
+        The AOS residual in arcseconds
+
+    Returns
+    -------
+    float or np.ndarray
+        The corrected AOS residual
+    """
+    return 1.06 * np.log(1 + aos_resid)
+
+
+def calcAOSResid(
+    zernikes: np.ndarray,
+    diameter: float = 8.36,
+    obscuration: float = 0.612,
+    jmin: int = 4,
+) -> float | np.ndarray:
+    """Calculate the AOS residual for the wavefront error.
+
+    The AOS residual quantifies how much the wavefront error contributes
+    in quadrature to the PSF FWHM
+
+    Parameters
+    ----------
+    zernikes : np.ndarray
```

Comment:
I'm assuming when using sparse Zernikes you should fill in with 0s the unused indices? I think it would be good to mention that here.